File: core/c/registerOneliners.c
// C port of core/swift/registerOneliners.swift
#include "dialect.h"

// Register several oneliner callbacks to a controller.
// items is an array of { field, cb } pairs mirroring the Swift
// alternating [field, callback, ...] payload.
void dialectRegisterOneliners(DialectController* ctrl, const DialectOneliner* items, size_t count) {
    for (size_t i = 0; i < count; i++) {
        dialectControllerRegisterFieldCallback(ctrl, items[i].field, items[i].cb);
    }
}

File: core/nim/registerOneliners.nim
# Nim port of core/swift/registerOneliners.swift

import dialectController

type
  DialectOneliner*[T] = object
    field*: string
    cb*: proc(c: T)

# Register several oneliner callbacks (items = [field, cb, ...])
proc dialectRegisterOneliners*[T](
  ctrl: DialectController[T],
  items: openArray[DialectOneliner[T]],
) =
  for item in items:
    registerFieldCallback(ctrl, item.field, item.cb)